server/public/plugin/context_rpc_test.go

```
@@ -0,0 +1,419 @@
+	serverConnection, clientConnection := net.Pipe()
+	server := rpc.NewServer()
+	require.NoError(t, server.RegisterName("Plugin", &successfulPluginHTTPStreamRPCServer{}))
+	go server.ServeConn(serverConnection)
+
+	client := rpc.NewClient(clientConnection)
+	defer client.Close()
+	body := &blockingReadCloser{closed: make(chan struct{})}
+	request, err := http.NewRequestWithContext(context.Background(), http.MethodPost, "/plugin/path", body)
+	require.NoError(t, err)
+
+	response, err := (&apiRPCClient{client: client, muxBroker: muxBroker}).pluginHTTPStream(request, false)
```

golangci-lint (2.13.2)

[error] response body must be closed

(bodyclose)